crates/broker/src/order_picker.rs

```diff
@@ -218,25 +258,27 @@ where
             format_units(gas_price, "gwei").unwrap()
         );
 
-        if order_gas_cost > order.request.offer.maxPrice {
+        if order_gas_cost > order.request.offer.maxPrice && !lock_expired {
+            // Cannot check the gas cost for lock expired orders where the reward is a fraction of the stake
+            // TODO: This can be added once we have a price feed for the stake token in gas tokens
             tracing::info!(
                 "Estimated gas cost to lock and fill order {order_id:x} execeeds max price; max price {}", format_ether(order.request.offer.maxPrice)
             );
             self.db.skip_order(order_id).await.context("Failed to delete order")?;
-            return Ok(None);
+            return Ok(Skip);
         }
 
         if order_gas_cost > available_gas {
             tracing::warn!("Estimated there will be insufficient gas to lock and fill order {order_id:x} after locking and fulfilling pending orders; available_gas {} ether", format_ether(available_gas));
             self.db.skip_order(order_id).await.context("Failed to delete order")?;
-            return Ok(None);
+            return Ok(Skip);
```

Comment:
For lock expired orders, when we fulfill we don't actually need to lock, so the `order_gas_cost` variable on L252 ends up being an over-estimate, meaning we'll skip more orders than needed here. 

The logging here and above also mentions locking which could be confusing. Would be good to update it based on `lock_expired`

Reply:
altered that gas computation to skip the locking gas for a lock expired order, also just changed the warning message a little bit so it applies in either case